tools/modules/EnsEMBL/Web/Component/Tools/VEP/InputForm.pm

```diff
@@ -622,70 +624,72 @@ sub _build_additional_annotations {
 
   ## REGULATORY DATA
   $current_section = 'Regulatory data';
-
-
   my @regu_species = map { $_->{'value'} } grep {$hub->get_adaptor('get_EpigenomeAdaptor', 'funcgen', $_->{'value'})} grep {$_->{'regulatory'}} @$species;
 
-  my @regu_species_classes = map { "_stt_".$_ } @regu_species;
-
-  my $regu_class = (scalar(@regu_species_classes)) ? join(' ',@regu_species_classes) : '';
+  if(@regu_species) {
+    my @regu_species_classes = map { "_stt_".$_ } @regu_species;
 
-  $fieldset = $form->add_fieldset({'legend' => $current_section, 'no_required_notes' => 1, class => $regu_class});
+    my $regu_class = (scalar(@regu_species_classes)) ? join(' ',@regu_species_classes) : '';
 
-  for (@regu_species) {
-    # get available cell types
-    my $regulatory_build_adaptor = $hub->get_adaptor('get_RegulatoryBuildAdaptor', 'funcgen', $_);
-    my $regulatory_build = $regulatory_build_adaptor->fetch_current_regulatory_build;
-    my @cell_types = ();
-    foreach (sort {$a->short_name cmp $b->short_name} @{$regulatory_build->get_all_Epigenomes}) {
-      my $short_name = $_->short_name;
-      my $rm_white_space_label = $short_name;
-      $rm_white_space_label =~ s/ /\_/g;
-      push @cell_types, { value => $rm_white_space_label, caption => $short_name }; 
-    } 
+    $fieldset = $form->add_fieldset({'legend' => $current_section, 'no_required_notes' => 1, class => $regu_class});
 
-    $fieldset->add_field({
```

Comment:
Hard to understand this change. Probably a small thing but could we do it over a call if it is hard to explain it here?